### ISSUE TITLE
Work in Node.js

### ISSUE THREAD

**Issue by garrensmith** [garrensmith/abletonpush#6]
I'm not sure how much effort this would take but it would be nice to explore the idea and see if it is possible to get this library working in Node.js.

**Comment by codenakama** [comment]
@garrensmith what is the use case for this? currently webmidi is not supported in nodejs, however there is an attempt to that - [web-midi-api](https://www.npmjs.com/package/web-midi-api)

the other way is to just use midi through [usb](https://www.npmjs.com/package/usb) as a fallback in Node and write implementation for it. I am also aware of a node based implementation for Push [here](https://www.npmjs.com/package/ableton-push2)

**Comment by garrensmith** [comment]
It is more of an idea than an actual use case. If you writing a nodejs app could be useful. But this is definitely not the highest priority

________________________________

**Comment by codenakama** [comment]
@garrensmith I see, it makes sense. perhaps we could leverage from https://www.npmjs.com/package/ableton-push2 since it only offers api to communicate midi through lib-usb. Or fork it and add pieces we need or write our own implementation.

**Comment by garrensmith** [comment]
Yes. Nice idea. Lets focus on the other issues first and then come back to this one
